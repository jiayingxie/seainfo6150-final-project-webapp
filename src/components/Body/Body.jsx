import React, { Component } from 'react';
import { Link, Route } from 'react-router-dom';
import styles from './Body.module.css';
import ProgramList from '../ProgramList/ProgramList.jsx';

export default class Body extends Component {
    render() {
        const departmetnts = this.props.data;
        let navContent = (
            <div className={styles.navBar}>
                {
                    departmetnts.map(dep => <Link to={`/engineering/${dep.departmentID}`} key={dep.departmentID}>{dep.departmentName}</Link>)
                }
            </div>
        )
        let displayContent = (
            <div className={styles.displayShow}>
                {
                    departmetnts.map(dep => <Route path={`/engineering/${dep.departmentID}`} key={dep.departmentID}><ProgramList department={dep} /></Route>)
                }
            </div>
        )

        return (
            <>
                <header>
                    <nav>
                        {navContent}
                    </nav>
                </header>
                <div>
                    {displayContent}
                </div>
            </>
        )
    }
}
