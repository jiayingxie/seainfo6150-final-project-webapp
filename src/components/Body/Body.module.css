.navBar {
    display: flex;
    flex-wrap: wrap;
    justify-content: space-between;
    font-size: 16px;
    padding: 0 10px;
}
.navBar a{
    display: inline-block;
    
    padding: 5px;
    background-color: wheat;
    border-radius: 10px;

    color: black;
    font-size: 18px;
}

.displayShow{
    text-align: center;
}