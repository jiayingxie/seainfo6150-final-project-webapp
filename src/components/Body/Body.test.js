import React from "react";
import Body from "./Body";

describe("Body tests", () => {
    // the method is to get test data
    const fetchData = async () => {
        const response = await fetch(`https://demo2247515.mockable.io/engineering`);
        const responseJson = await response.json();
        return Object.values(responseJson);
    };
    it("renders correctly when passed a data prop", async () => {
        const fetchedData = await fetchData();
        const { container } = render(<Body data={fetchedData} />);
        expect(container).toMatchSnapshot();
    });
});