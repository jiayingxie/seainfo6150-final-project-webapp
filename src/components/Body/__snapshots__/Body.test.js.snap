// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Body tests renders correctly when passed a data prop 1`] = `
<div>
  <header>
    <nav>
      <div
        class="navBar"
      >
        <a
          href="/engineering/multidisciplinary"
        >
          multidisciplinary
        </a>
        <a
          href="/engineering/civilenvironmental"
        >
          civilenvironmental
        </a>
        <a
          href="/engineering/bioengineering"
        >
          bioengineering
        </a>
        <a
          href="/engineering/chemical"
        >
          chemical
        </a>
        <a
          href="/engineering/electricalcomputer"
        >
          electricalcomputer
        </a>
        <a
          href="/engineering/mechanicalindustrial"
        >
          mechanicalindustrial
        </a>
      </div>
    </nav>
  </header>
  <div>
    <div
      class="displayShow"
    />
  </div>
</div>
`;
